addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/FadingWiFiLEDDriver.java

```diff
@@ -143,50 +159,112 @@ private void changeState(final InternalLedState newState) throws IOException {
         }
     }
 
-    private void fadeToState(final InternalLedState newState) throws IOException {
-        if (!newState.equals(realTargetState)) {
-            keepFading = false;
-            realTargetState = newState;
+    final static class LEDFaderRunner implements Runnable {
+        protected Logger logger = LoggerFactory.getLogger(LEDFaderRunner.class);
+        InternalLedState fadeState;
+        int fadeStepper = 1;
+
+        final InternalLedState toState;
+        final InternalLedState fromState;
+        final Future<?>[] faderArray;
+        final int steps;
+        final Function<InternalLedState, Boolean> ledSender;
+
+        public LEDFaderRunner(Future<?>[] faderFuture, InternalLedState fromState, InternalLedState toState, int steps, Function<InternalLedState,  Boolean> ledSender) {
+            this.faderArray = faderFuture;
+            fadeState=fromState;
```

Comment:
> Idea
Do you mean Intellij?

If you use the preconfigured Eclipse for openHAB you should just be able to call https://stackoverflow.com/a/15655127/3656774

For Intellij you can export it from the predefined Eclipse and import that in Intellij. For other IDE's it might depend.

But you could also just add spaces around the = 😉